src/chainrules.jl

```diff
@@ -6,3 +6,122 @@ ChainRulesCore.@scalar_rule(
     ),
     (x, - tanh(Ω + b) * x, x - 1),
 )
+
+# `OrderedBijector`
+function ChainRulesCore.rrule(::typeof(_transform_ordered), y::AbstractVector)
+    function _transform_ordered_adjoint(Δ)
+        Δ_new = similar(y)
+        n = length(Δ)
+        @assert n == length(Δ_new)
+
+        s = sum(Δ)
+        Δ_new[1] = s
+        @inbounds for i in 2:n
+            # Equivalent to
+            #
+            #    Δ_new[i] = sum(Δ[i:end]) * yexp[i - 1]
+            #
+            s -= Δ[i - 1]
+            Δ_new[i] = s * exp(y[i])
+        end
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    return _transform_ordered(y), _transform_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_ordered), y::AbstractMatrix)
+    function _transform_ordered_adjoint(Δ)
+        Δ_new = similar(y)
+        n = length(Δ)
+        @assert n == size(Δ_new, 1)
+
+        s = sum(Δ)
+        Δ_new[1, :] .= s
+        @inbounds for i in 2:n
+            # Equivalent to
+            #
+            #    Δ_new[i] = sum(Δ[i:end]) * yexp[i - 1]
+            #
+            s -= Δ[i - 1]
+            Δ_new[i, :] = s * exp.(y[i, :])
+        end
+
+        return (ChainRulesCore.NoTangent(), Δ_new)
+    end
+
+    return _transform_ordered(y), _transform_ordered_adjoint
+end
+
+function ChainRulesCore.rrule(::typeof(_transform_inverse_ordered), x::AbstractVector)
+    r = similar(x)
+    @inbounds for i = 1:length(r)
+        if i == 1
+            r[i] = 1
+        else
+            r[i] = x[i] - x[i - 1]
+        end
+    end
```

Comment:
Why not just `r = diff(x)` and handle the first case separately below?

Reply:
I did this because it makes the code for the adjoint a bit "simpler" by allowing us to use the same index for both `Δ` and `r`, while if we do something like `diff` we need to adjust for this. It just makes it slightly more confusing to read IMO. Plus we need to handle the case of `i = 1` in the computation which also reads a bit worse IMO.